Sprinkle of Yeezus/Settings.swift

```diff
@@ -11,52 +11,68 @@ import UIKit
-        } else {
-            DescLabel.isHidden = true
-            DescButton.isHidden = true
-            notificationsOn = false
-        }
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        sprinkleNotifications(sprinkleList)
     }
     
-    @IBOutlet weak var TimePicked: UIDatePicker!
-    var sprinkleTimePicked = Date()
-    @IBAction func TimePicker(_ sender: UIDatePicker){
-        sprinkleTimePicked = TimePicked.date
+    // MARK: - IBAction
+    @IBAction private func NotificationSwitch(_ sender: UISwitch) {
+        if sender.isOn {
+            notificationCenter.requestAuthorization(options: [.alert, .sound], completionHandler: { didAllow, error in
+                DispatchQueue.main.async {
+                    sender.isOn = didAllow
+                    self.updateUI()
+                }
+            })
+        }
+        
+        updateUI()
     }
     
-    func sprinkleNotifications(_ sprinkleList: Array<Sprinkle>) {
-        if notificationsOn == true{
-            
-            let notification = UNMutableNotificationContent()
-            let notificationText = pickRandomQuote(sprinkleList)
-            notification.body = " \"\(notificationText.quote)\" \(notificationText.quoteSource), \(notificationText.date)"
-            
-            let timeForSprinkle = TimePicked.date
-            let components = Calendar.current.dateComponents([.hour, .minute], from: timeForSprinkle)
-            InfoLabel.text = "Sprinkles will be sent every day at \(components.hour):\(components.minute)"
-            let notificationTime = UNCalendarNotificationTrigger(dateMatching: components, repeats: true)
-            
-        }
+    @IBAction private func TimePicker(_ sender: UIDatePicker){
+        sprinkleTimePicked = sender.date
+    }  
+    
+    // MARK: - Private
+    private func updateUI() {
+        [DescLabel, DescButton].forEach { $0.isHidden = !Switch.isOn }
     }
     
-    override func viewDidLoad() {
-        sprinkleNotifications(sprinkleList)
+    private func sprinkleNotifications(_ sprinkleList: Array<Sprinkle>) {
+        notificationCenter.removeAllPendingNotificationRequests()
+        
+        guard Switch.isOn else {
+            return
+        }
+        
+        let notification = UNMutableNotificationContent()
+        let notificationText = pickRandomQuote(sprinkleList)
+        notification.body = "\"\(notificationText.quote)\" \(notificationText.quoteSource), \(notificationText.date)"
+
+        let components = Calendar.current.dateComponents([.hour, .minute], from: sprinkleTimePicked)
+        let hour = components.hour ?? 12
+        let minute = components.minute ?? 0
+        
+        InfoLabel.text = "Sprinkles will be sent every day at \(hour):\(minute)"
+        
+        let trigger = UNCalendarNotificationTrigger(dateMatching: components, repeats: true)
+        let request = UNNotificationRequest(identifier: "Sprinkle-of-Yeezus", content: notification, trigger: trigger)
```

Comment:
Using the trigger you created, I added the creation of the official `UNNotificationRequest`. I gave it just a simple identifier (you can use different identifiers to separate different types of notifications, which might not be necessary for this app if you are only sending Sprinkles).

Reply:
Thanks so much for you help. I'm still a noob at github and was wondering.. Are the red minuses lines you deleted and the green plus signs lines that you added? It took me a second to realize what was going on. Also, I was up last night trying to figure out how to give the hour and minute default values and the way you did it makes much more sense. It's probably easy to see here that I did better at making the rest of the app and when it came to this I was basically googling and hoping for the best.

Reply:
Yup! Red are either deletions, or what the code previously was. And then Green are either additions or what the code has been changed to.

And totally! The `??` is called a "Nil Coalescing Operator", which basically is a fancy term for giving a default value to an optional.

Feel free to comment/message me with anything you might want to know more about, even before Google. Sometimes StackOverflow provides an answer, but it might not be the best solution.

Reply:
Also, not sure if you are new to Git, do you want an explanation about this whole Pull Request and Code Review process?